source/user/mcscf.rst

```diff
@@ -1,57 +1,299 @@
+The user can "manually" select the MO orbital indices (in a 1-based indexing scheme) and pass them to the ``sort_mo`` function.
+  See :source:`examples/mcscf/10-define_cas_space.py` for more details.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 4, 4)
+  # Note sort_mo by default take the 1-based orbital indices.
+  mo = mycas.sort_mo([5,6,8,9])
+  mycas.kernel(mo)
+
+
+3) Specifying the number of orbitals in each symmetry group. 
+This strategy can occasionally be helpful when the initial guess orbitals are not easily identifiable.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(mf, 12, 12)
+  ncore = {'A1g':5, 'A1u':5}
+  ncas = {'A1g':2, 'A1u':2,'E1ux':1, 'E1uy':1, 'E1gx':1, 'E1gy':1,
+              'E2ux':1, 'E2uy':1, 'E2gx':1, 'E2gy':1}
+  mo = mcscf.sort_mo_by_irrep(mycas, mf.mo_coeff, ncas, ncore)
+  mycas.kernel(mo)
+
+A similar approach where we specify the electron occupations by irreducible representation is also possible by setting ``mycas.fcisolver.irrep_nelec``.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 8, 8)
+  mycas.fcisolver.irrep_nelec = {"A1g": (2, 1), "A1u": (1, 1), "E1ux": (1, 1), "E1uy": (1, 0)}
+
+4) Use automated strategies (``avas`` and ``dmet_cas``) to pick an active space based on AO orbitals you're targeting.
+For more details, see :source:`examples/mcscf/43-avas.py` and :source:`examples/mcscf/43-dmet_cas.py`.
+
+.. code-block:: python
+
+  from pyscf.mcscf import avas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, orbs = avas.avas(mf, ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+
+  
+
+.. code-block:: python
+
+  from pyscf.mcscf import dmet_cas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, mo = dmet_cas.guess_cas(mf, mf.make_rdm1(), ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+  mycas.kernel(mo)
+
+
+Frozen Core MCSCF
+-----------------
+
+To reduce to computational expense of CASSCF calculations, users can "freeze" orbitals thereby excluding them from optimization.
+
+Users can specify the number of lowest orbitals to freeze:
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 6, 8)
+  mycas.frozen = 2
+  mycas.kernel()
+
+
+Users can also specify a list of orbital indices (0-based).
+These may be occupied, virtual, or active orbitals.
+
+.. code-block:: python
+  mycas = mcscf.CASSCF(myhf, 6, 8)
+  mycas.frozen = [0,1,26,27]
+  mycas.kernel()
+
+See :source:`examples/mcscf/19-frozen_core.py` for a complete example.
+
+
+State-Averaged Calculations
+---------------------------
+
+When dealing with states that are close in energy, it can be helpful to perform state average calculations where the multireference orbitals are optimized for multiple states.
+The ``state_average_`` function (note the hanging underscore) is a member function of ``CASCI``/``CASSCF`` objects and takes the weights of the states as input.
+The weights can be any normalized and non-negative array of values.
```

Comment:
Nevertheless,most widely state averaging with equal weights is used.

Reply:
```suggestion
Equal weights are often, but not always, used.
 ```

Reply:
What if the weights given in are not normalized?

Say, if the user inputs an array `[1 1 2 3 5 8]`, does the program use these weights as is, normalize them, or what?

Reply:
I also wonder why negative weights are not allowed.

Reply:
I don't think negative values wouldn't make sense when you calculate the energy using these weights see the docs here https://github.com/pyscf/pyscf/blob/7be5e015b2b40181755c71d888449db936604660/pyscf/mcscf/addons.py#L788

If you chose all negative weights your energy would flip sign.

There is an assertion check see below:
https://github.com/pyscf/pyscf/blob/7be5e015b2b40181755c71d888449db936604660/pyscf/mcscf/addons.py#L808